backend/Infrastructure/Repositories/EngagementDbRepository.cs

```diff
@@ -0,0 +1,23 @@
+using Core.DomainModels;
+using Core.IRepositories;
+using Database.DatabaseContext;
+using Microsoft.EntityFrameworkCore;
+
+namespace Database.Repositories;
+
+public class EngagementDbRepository : IEngagementRepository
+{
+    private readonly ApplicationContext _context;
+
+    public EngagementDbRepository(ApplicationContext context)
+    {
+        _context = context;
+    }
+
+    public Engagement? GetEngagementById(int id)
+    {
+        // Her trenger jeg en bedre måte å deale med relasjoner: Ideelt kanskje et customer-repository på et vis? 
+        // How to relasjoner i repository?
+        return _context.Project.Include(p => p.Customer).SingleOrDefault(p => p.Id == id);
+    }
```

Comment:
`.Include()` er helt riktig måte :D
```suggestion
    public Engagement? GetEngagementById(int id, CancellationToken cancellationToken)
    {
        return _context.Project
            .AsNoTracking()
            .Include(p => p.Customer)
            .FirstOrDefaultAsync(p => p.Id == id, cancellationToken);
    }
```

Reply:
`.SingleOrDefault` kan være bedre enn `.FirstOrDefault` utifra atferden man har lyst på, men lurt å huske på at `.Single...` thrower hvis mer enn ett element matcher.

Reply:
Det er også gode vibes å forwarde `cancellationToken` nedover, hvis man har noen tunge jobber eller appen skrus av så stopper ting nedover på en litt mer elegant måte enn å kræsje i veggen

Reply:
Så alle interfacene til `Repositories` kan være `async` og få med seg en `CancellationToken` som siste argument

Reply:
Men på om relasjoner vi trenger eller ikke skal være med - Skal vi ha ymse ulike ting som `GetEngagementWithProject` osv? Her er det jo tilfeldig at vi tar med Customer egentlig